heron/tmaster/src/cpp/manager/tmaster.cpp

```diff
@@ -461,12 +583,16 @@ void TMaster::DeActivateTopology(VCallback<proto::system::StatusCode> cb) {
 }
 
 void TMaster::CleanAllStatefulCheckpoint() {
-  // TODO(nlu): to be implemented later
+  ckptmgr_client_->SendCleanStatefulCheckpointRequest("", true);
+}
+
+void TMaster::HandleStatefulCkptSave(std::string _oldest_ckpt) {
+  ckptmgr_client_->SendCleanStatefulCheckpointRequest(_oldest_ckpt, false);
 }
 
 // Called when ckptmgr completes the clean stateful checkpoint request
 void TMaster::HandleCleanStatefulCheckpointResponse(proto::system::StatusCode _status) {
-  // TODO(nlu): to be implemented later
+  controller_->HandleCleanStatefulCheckpointResponse(_status);
```

Comment:
nitpick: s/controller_/tmater_controller_ to differentiate with the stateful_controller_